client/src/components/LibraryNavigator/LibraryModel.ts

```diff
@@ -54,30 +54,105 @@ class LibraryModel {
 
   public getTableResultSet(item: LibraryItem): PaginatedResultSet<TableData> {
     return new PaginatedResultSet<TableData>(
-      async (start: number) => {
+      async (start: number, end: number) => {
         await this.setup();
+        const limit = end - start + 1;
         return await this.retryOnFail(
           async () =>
             await this.dataAccessApi.getRows(
               {
                 sessionId: this.sessionId,
                 libref: item.library || "",
                 tableName: item.name,
-                includeColumnNames: true,
+                includeIndex: true,
                 start,
-                limit: 100,
+                limit,
               },
               requestOptions,
             ),
         );
       },
       (response) => ({
-        headers: response.data.items[0],
-        rows: response.data.items.slice(1),
+        rows: response.data.items,
+        count: response.data.count,
       }),
     );
   }
 
+  public async getTableContents(item: LibraryItem) {
+    await this.setup();
+    let start = 0;
+    let items: TableData["rows"] = [];
+    let headers: { columns: string[] } | undefined;
+    let totalItemCount = Infinity;
+    do {
+      const { data } = await this.retryOnFail(
+        async () =>
+          await this.dataAccessApi.getRowsAsCSV(
+            {
+              includeColumnNames: true,
+              includeIndex: true,
+              libref: item.library || "",
+              limit: DefaultRecordLimit,
+              sessionId: this.sessionId,
+              start,
+              tableName: item.name,
+            },
+            requestOptions,
+          ),
+      );
+
+      headers = data.items.shift();
+      items = [...items, ...data.items];
+      totalItemCount = data.count;
+      start += DefaultRecordLimit;
+    } while (start < totalItemCount);
```

Comment:
nit: I'm not sure if we can fetch multiple pages concurrently and if it can improve efficiency for some large tables.
BTW., I wrote a throttle util when fetching run results
https://github.com/sassoftware/vscode-sas-extension/blob/main/client/src/components/utils.ts#L27

Reply:
This could be interesting. I vote for deferring this to another PR if/when it becomes an issue. That sound good? I think it could work, but we do rely on `data.count` for total item count. I suppose we could just making requests until we get no items, though.

On second thought, I'll take a stab at it today to see how much of a challenge it will be.

Reply:
That was pretty straightforward. Done.